src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4774,6 +4782,18 @@ internal bool CheckValEscape(SyntaxNode node, BoundExpression expr, SafeContext
                     }
                     return true;
 
+                case BoundKind.ObjectOrCollectionValuePlaceholder:
+                    if (_placeholderScopes?.TryGetValue((BoundObjectOrCollectionValuePlaceholder)expr, out var scope) == true)
+                    {
+                        if (!scope.IsConvertibleTo(escapeTo))
+                        {
+                            Error(diagnostics, inUnsafeRegion ? ErrorCode.WRN_EscapeVariable : ErrorCode.ERR_EscapeVariable, node, expr.Syntax);
```

Comment:
>inUnsafeRegion ? ErrorCode.WRN_EscapeVariable [](http://example.com/codeflow?start=47&length=45)

Are we testing the `unsafe { }` region case?

Reply:
Added `unsafe` test. I'm not sure the `WRN_EscapeVariable` here is reachable (at least in the tested scenarios, this is used with discarded diagnostics just for the return value), but at least it's consistent with other similar places (and I imagine it might be reachable with more complex nested scenarios).